hydra_slayer/search_path.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, List, Optional, Sequence, Union
+from dataclasses import dataclass
+import pydoc
+
+SearchPathEntryDescription = Union[str, Sequence[str]]
+
+
+@dataclass
+class SearchPathEntry:
+    """
+    A single search path entry. Consists of an actual path to a Python object and its alias.
+    If alias wasn't provided then the last dot-separated part of path is treated as an alias.
+    """
+
+    path: str
+    alias: str
+
+    def is_referenced_by(self, name: str) -> bool:
+        """
+        Args:
+            name: An import path to some Python object
+
+        Returns:
+            True: when this SearchPathEntry corresponds to an object referenced by name
+            or to its top-level parent module.
+            False: otherwise
+        """
+        name_prefix = name.split(".")[0]
+        return name_prefix == self.alias
+
+    def resolve(self, name: str) -> str:
+        """
+        Resolve path to a python object referenced by name.
+
+        Args:
+            name: an import path to a Python object
+
+        Returns:
+            A full path to a Python object resolved with this SearchPathEntry
+
+        Raises:
+            ValueError: When name can't be resolved with this instance of SearchPathEntry.
+                Check it with SearchPathEntry.is_referenced_by before
+        """
+        if not self.is_referenced_by(name):
+            raise ValueError(
+                f"Can't resolve {name}. It is not referenced by the current SearchPathEntry"
+            )
+
+        name = name.split(".")
+        return ".".join([self.path] + name[1:])
+
+    @staticmethod
+    def from_description(description: SearchPathEntryDescription) -> "SearchPathEntry":
```

Comment:
Currently we can create `SearchPathEntry` from `str` or from `Sequence[str]` of length 2.

But in .yaml files nested lists look kinda ugly. Config would look like this if we wanted to refer to `numpy` with `np`.
```
_imports_:
- - numpy
   - np
```

Since Python packages can't contain a space in its name we could express the same thing with a single string e.g.:
```
_imports_:
  - numpy np
```

or

```
_imports_:
  - numpy as np
```

What do you think? Would this solution be better?